pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -291,7 +290,6 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env *model.En
 		// Construct the default filter chain.
 		if len(allChains) == 0 {
 			log.Infof("Use default filter chain for %v", endpoint)
-			allChains = []plugin.FilterChain{plugin.FilterChain{}}
```

Comment:
don't we still need this? it adds the default filter chain

Reply:
Ah, good catch. Yes, I think we'll still need this line (so that we enter the loop below and add the HTTP/TCP config w/ default TLS, match, etc).

Reply:
FWIW I'd expect tests to fail due to this if we do indeed need this. Given that the e2e tests are failing for reachability, we could be hitting this condition (unclear to me how healthy the e2e tests are atm).

Reply:
The default is set here.
https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/listener.go#L255
Do I miss something?

Reply:
Oh, IC.  `allChains = []plugin.FilterChain{plugin.FilterChain{}}` indicates length 1 slice.

Reply:
Add a comment to the effect of "add one empty entry to the list so we generate a default listener below" please.

Reply:
ok

Reply:
done